csharp/src/Ice/Properties.cs

```diff
@@ -348,14 +348,8 @@ public void setProperty(string key, string val)
             throw new InitializationException("Attempt to set property with empty key");
         }
 
-        // Find the property, log warnings if necessary
-        Property? prop = findProperty(key, true);
-
-        // If the property is deprecated by another property, use the new property key
-        if (prop != null && prop.deprecatedBy != null)
-        {
-            key = prop.deprecatedBy;
-        }
+        // Checks if the property is a known Ice property and logs warnings if necessary
+        findProperty(key, true);
```

Comment:
calling a "findXxx" for side effects is not very nice.

Reply:
I agree. We could rename it to something like `checkProperty`, but I'm note sure that's much better. 

We can only split one of the warnings out of the `findProperty` method. The other two depend on the knowledge that we have a known Ice prefix or not.

Any ideas?